INSTALL

```diff
@@ -95,22 +95,15 @@ The list of available extras:
 
 # START EXTRAS HERE
 airbyte, alibaba, all, all_dbs, amazon, apache.atlas, apache.beam, apache.cassandra, apache.drill,
-apache.druid, apache.hdfs, apache.hive, apache.kylin, apache.livy, apache.pig, apache.pinot,
-apache.spark, apache.sqoop, apache.webhdfs, arangodb, asana, async, atlas, atlassian.jira, aws,
-azure, cassandra, celery, cgroups, cloudant, cncf.kubernetes, common.sql, crypto, dask, databricks,
-datadog, dbt.cloud, deprecated_api, devel, devel_all, devel_ci, devel_hadoop, dingding, discord,
-doc, docker, druid, elasticsearch, exasol, facebook, ftp, gcp, gcp_api, github, github_enterprise,
-google, google_auth, grpc, hashicorp, hdfs, hive, http, imap, influxdb, jdbc, jenkins, kerberos,
-kubernetes, ldap, leveldb, microsoft.azure, microsoft.mssql, microsoft.psrp, microsoft.winrm, mongo,
-mssql, mysql, neo4j, odbc, openfaas, opsgenie, oracle, pagerduty, pandas, papermill, password,
-pinot, plexus, postgres, presto, qds, qubole, rabbitmq, redis, s3, salesforce, samba, segment,
-sendgrid, sentry, sftp, singularity, slack, snowflake, spark, sqlite, ssh, statsd, tableau, tabular,
-telegram, trino, vertica, virtualenv, webhdfs, winrm, yandex, zendesk
-# END EXTRAS HERE
-
-# For installing Airflow in development environments - see CONTRIBUTING.rst
-
-# COMPILING FRONT-END ASSETS (in case you see "Please make sure to build the frontend in static/ directory and then restart the server")
-# Optional : Installing yarn - https://classic.yarnpkg.com/en/docs/install
-
-python setup.py compile_assets
```

Comment:
It seems like these lines were accidentally removed.

Reply:
its not accidentally. last CI ran to a problem to fix it I needed to ran this command:
```shell
breeze setup regenerate-command-images
```
and it changed some files.
https://github.com/apache/airflow/actions/runs/3380995761/jobs/5614846732

Reply:
IIRC we merged `compile_assets` into startup so this is no longer needed? Don’t quite remember, I think @potiuk might?

Reply:
Good eye @bdsoha - I think this was removed by accident early when generating extras (not really connected with regenerating images). Could you please restore it @blcksrx ?

@uranusjr: not really.

A bit of context.

The `compile_assets` is only automated when you start breeze, not in any other circumstances. The INSTALL here describes the process of installing airflow manualy from sources - this is the absolute requirement of the Apache Software Foundation, that "sufficient platform and tools" you should be able to take sources published in https://downloads.apache.org/airflow/2.4.3/ as *-sources.tar.gz and you should be able to get workinf software.

This is what INSTALL doc is for (it is also the first thing anyone who takes the .source.tgz file should look at). It is also described in this part of our installation documentation https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html#using-released-sources. Those published sources should only contain source files that can and should be possible to manually edit to make any change to the software of ours. It should not contain any compiled sources 

This is described here: https://www.apache.org/legal/release-policy.html#source-packages

> Every ASF release MUST contain one or more source packages, which MUST be sufficient for a user to build and test the release provided they have access to the appropriate platform and tools. A source release SHOULD not contain compiled code.

This is not optional, it's a mandatory requirement of ASF release policy and we MUST follow it - otherwise PMC members of Apache Airlfow are not idemnified by the Apache Sofware Foundation in case of any legal battles - this is one of the reasons why we have release voting process and why PMC members only have binding votes.

In this context, running `python setup.py compile_assets` is absolutely necessary step in those instructions - we cannot (and we do not) have compiled assets released in the source.tgz. file and INSTALL document is about this file.

BTW. By answering this one I noticed that there are few things about installing providers and using constraints that are not needed here, actually, so I will shortly correct it.